RTOS.c

```diff
@@ -131,52 +190,59 @@ void rtosInit(void){
 	TCBList[MAIN_TASK_ID].state = RUNNING;
 	runningTCB = &(TCBList[MAIN_TASK_ID]);
 
-	//initialize ready list to NULL
-	readyListHead = NULL;
+	//initialize Priority Queues
+	for (taskPriority_t priority = HIGHEST_PRIORITY; priority < NUM_PRIORITIES; priority++){
+		readyTaskPriorityQueue[priority] = NULL;
```

Comment:
this should be 
`readyTaskPriorityQueue[priority]->head = NULL;
readyTaskPriorityQueue[priority]->tail = NULL;`
same for `waitingTaskPriorityQueue[priority]`